src/Features/VisualBasic/Portable/InlineParameterNameHints/InlineParameterNameHintsService.vb

```diff
@@ -0,0 +1,78 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.InlineParameterNameHints
+
+    <ExportLanguageService(GetType(IInlineParameterNameHintsService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class InlineParameterNameHintsService
+        Implements IInlineParameterNameHintsService
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Public Async Function GetInlineParameterNameHintsAsync(document As Document, textSpan As TextSpan, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of InlineParameterHint)) Implements IInlineParameterNameHintsService.GetInlineParameterNameHintsAsync
+            Dim tree = Await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(False)
+            Dim root = Await tree.GetRootAsync(cancellationToken).ConfigureAwait(False)
+            Dim spans = New List(Of InlineParameterHint)
+
+            Dim semanticModel = Await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(False)
+            Dim nodes = root.DescendantNodes()
+
+            For Each node In nodes
+                If TypeOf node Is SimpleArgumentSyntax Then
+                    Dim simpleArgument = TryCast(node, SimpleArgumentSyntax)
```

Comment:
In this case, just do the TryCast first and check that for being null; that saves an extra type check.